src/store/playlist-library/helpers.ts

```diff
@@ -0,0 +1,168 @@
+  PlaylistLibrary,
+  PlaylistLibraryFolder,
+  PlaylistLibraryIdentifier
+} from 'models/PlaylistLibrary'
+
+/**
+ * Finds an item by id in the playlist library
+ * @param library
+ * @param playlistId
+ * @returns the identifier or false
+ */
+export const findInPlaylistLibrary = (
+  library: PlaylistLibrary | PlaylistLibraryFolder,
+  playlistId: ID | SmartCollectionVariant
+): PlaylistLibraryIdentifier | false => {
+  if (!library.contents) return false
+
+  // Simple DFS (this likely is very small, so this is fine)
+  for (const item of library.contents) {
+    switch (item.type) {
+      case 'folder': {
+        const contains = findInPlaylistLibrary(item, playlistId)
+        if (contains) return contains
+        break
+      }
+      case 'playlist':
+      case 'explore_playlist':
+        if (item.playlist_id === playlistId) return item
+        break
+    }
+  }
+  return false
+}
+
+/**
+ * Finds the index of a playlist id in the library, returning false if not found
+ * @param library
+ * @param playlistId
+ * @returns {number | false}
+ */
+export const findIndexInPlaylistLibrary = (
+  library: PlaylistLibrary | PlaylistLibraryFolder,
+  playlistId: ID | SmartCollectionVariant
+): number | false => {
+  if (!library.contents) return false
+
+  // Simple DFS (this likely is very small, so this is fine)
+  for (const [i, item] of library.contents.entries()) {
+    switch (item.type) {
+      case 'folder': {
+        // TODO support folders. Need to devise a better system reorders
+        break
+      }
+      case 'playlist':
+      case 'explore_playlist':
+        if (item.playlist_id === playlistId) return i
+        break
+    }
+  }
+  return false
+}
+
+/**
+ * Removes a playlist from the library and returns the removed item as well as the
+ * updated library (does not mutate)
+ * @param library
+ * @param playlistId the id of the playlist to remove
+ * @returns { library, removed }
+ */
+export const removeFromPlaylistLibrary = (
+  library: PlaylistLibrary | PlaylistLibraryFolder,
+  playlistId: ID | SmartCollectionVariant
+): {
+  library: PlaylistLibrary | PlaylistLibraryFolder
+  removed: PlaylistLibraryIdentifier | null
+} => {
+  if (!library.contents) return { library, removed: null }
+
+  const newContents: (PlaylistLibraryFolder | PlaylistLibraryIdentifier)[] = []
+  let removed: PlaylistLibraryIdentifier | null = null
+  for (const item of library.contents) {
+    let newItem: PlaylistLibraryFolder | PlaylistLibraryIdentifier | null = item
+    switch (item.type) {
+      case 'folder': {
+        const res = removeFromPlaylistLibrary(item, playlistId)
+        removed = res.removed
+        newItem = {
+          type: item.type,
+          name: item.name,
+          contents: res.library.contents
+        }
+        break
+      }
+      case 'playlist':
+      case 'explore_playlist':
+        if (item.playlist_id === playlistId) {
+          removed = item
+          newItem = null
+        }
+    }
```

Comment:
nit. may want to add a break statement to avoid unintended behavior in the future